src/rust/engine/src/nodes.rs

```diff
@@ -1599,7 +1598,7 @@ impl Node for NodeKey {
       (NodeKey::Task(ref t), NodeOutput::Value(ref v)) if t.task.engine_aware_return_type => {
         let gil = Python::acquire_gil();
         let py = gil.python();
-        EngineAwareReturnType::is_cacheable(py, v).unwrap_or(true)
+        EngineAwareReturnType::is_cacheable(v.clone_ref(py).into_ref(py)).unwrap_or(true)
```

Comment:
```suggestion
        EngineAwareReturnType::is_cacheable(v.as_ref(py)).unwrap_or(true)
```
This will be more efficient, if the borrow checker allows it.
(Same for other uses of `into_ref` )

Reply:
Oh you're right. Had to do `(**v).as_ref()` though because this was a `&Value`, where `Value` is `Arc<PyObject>`